tests/test_net.py

```diff
@@ -0,0 +1,52 @@
+import unittest
+import tensorflow as tf
+import numpy as np
+
+from dual_net import DualNet
+
+class TestPrediction(unittest.TestCase):
+    def setUp(self):
+        self.sess = tf.Session()
+        self.net = DualNet(self.sess)
+        self.piece_net = DualNet(self.sess, action_rep='piece')
+
+        self.sess.__enter__()
+        tf.global_variables_initializer().run()
+
+        self.boards = np.random.random_sample([10, 8, 8, 13])
+        self.z = np.random.random_sample([10])
+
+    def test_predict(self):
+        policy, value = self.net.predict(self.boards)
+        self.assertEqual(policy.shape, (10, 64*64))
+
+    def test_train(self):
+        pi = np.random.random_sample([10, 64*64])
+        # violates unit testing protocol, but w/e
```

Comment:
So sometimes these tests pass and sometimes they fail? '

I wonder if it's worth just doing one run, computing `self.net.loss, self.net.regularization_loss` and a prediction, and then just asserting that their shapes are all as expected... 

Do you think that asserting that the loss goes down will be something we'll want to test frequently?